drake/doc/from_source.rst

```diff
@@ -71,16 +71,67 @@ Before running build, you will need to follow the instructions for your host sys
 
 Build the collection
 ====================
+There are two supported ways to build Drake:
+`Ninja <https://ninja-build.org/>`_ (recommended) and
+`Make <https://www.gnu.org/software/make/>`_ (legacy).
+Because the Drake build is not entirely out-of-source, you must pick one build
+system and stick to it. Switching build systems within the same working tree
+will produce CMake errors.
+
+.. _build_with_ninja:
+
+Build with Ninja (Recommended)
+------------------------------
+First, confirm that `Ninja <https://ninja-build.org/>`_ is installed
+on your system:
+
+::
+
+    ninja --version
+
+Drake uses `CMake <https://cmake.org/>`_ to generate Ninja files within an
+out-of-source build directory.
+
+::
+
+    mkdir drake-build
+    cd drake-build
+    cmake path/to/drake-distro -G Ninja
+    ninja
+
+Most ``cmake`` flags will work with Drake. For example,
+``CMAKE_BUILD_TYPE`` can be used to control the debug level.
+
+::
+
+    cmake path/to/drake-distro -G Ninja -DCMAKE_BUILD_TYPE:STRING=[Debug|Release|RelWithDebInfo]
+    ninja -j <num_CPUs>

```

Comment:
Ninja defaults to parallel builds, so unless you wish to limit the number of cores, you can safely omit `-j`.